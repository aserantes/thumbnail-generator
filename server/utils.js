const config = require("./config");

exports.hasValidSize = function (fileSize) {
  return fileSize !== 0 && fileSize < config.maxFileSize;
};

exports.hasValidType = function (fileType) {
  return config.validFileTypes.some((validFileType) =>
    validFileType.includes(fileType)
  );
};

exports.hasTypeMismatch = function (fileExt, fileType) {
  return !fileExt.includes(fileType);
};

exports.hasValidThumbSizes = function (array) {
  if (!Array.isArray(array)) {
    return false;
  } else {
    return array.every(
      (item) =>
        item.w &&
        typeof item.w === "number" &&
        item.h &&
        typeof item.h === "number"
    );
  }
};

exports.hasValidImageFitMode = function (imageFitMode) {
  return config.imageFitModes.includes(imageFitMode);
};

exports.colorLog = function (text, color) {
  let selectedColor = config.logColor[color];
  console.log(selectedColor + "%s\x1b[0m", text);
};
