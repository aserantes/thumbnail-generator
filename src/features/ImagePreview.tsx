import React from "react";
import {
  ButtonRow,
  ImageInfo,
  ImageInfoTextBlock,
  ImageInfoTextItem,
  ImagePreviewWrapper,
  Thumbnail,
} from "../components/Common";
import { ProcessButton, RestartButton } from "../components";
import { getName, getPath, getSize, getType } from "../store/selectors";
import { useSelector } from "react-redux";

export function ImagePreview() {
  const name = useSelector(getName);
  const type = useSelector(getType);
  const size = useSelector(getSize);
  const path = useSelector(getPath);
  const sizeText = `${(size / 1024).toFixed(2)} Kb`;

  return (
    <ImagePreviewWrapper data-testid="ImagePreview-wrapper">
      <ImageInfo>
        <Thumbnail alt="Image Preview" src={path} />
        <ImageInfoTextBlock>
          <ImageInfoTextItem>
            <strong>Name:&nbsp;</strong> {name}
          </ImageInfoTextItem>
          <ImageInfoTextItem>
            <strong>Type:&nbsp;</strong> {type}
          </ImageInfoTextItem>
          <ImageInfoTextItem>
            <strong>Size:&nbsp;</strong> {sizeText}
          </ImageInfoTextItem>
        </ImageInfoTextBlock>
      </ImageInfo>
      <ButtonRow>
        <RestartButton />
        <ProcessButton />
      </ButtonRow>
    </ImagePreviewWrapper>
  );
}
/*
const onSubmit = (e) => {
  e.preventDefault();
  const formData = new FormData();
  formData.append("file", file);

  // Pass a setUploadPercentage as callback here
  putData(formData, { onUploadProgress: setUploadPercentage )
    .then((response) => { 
      console.log(response);
    });
};
export function putData(formData, { onUploadProgress }) {
  return axios.put('/update', formData, {
    headers: {
      "Content-Type": "multipart/form-data",
    },
    onUploadProgress : (progressEvent) => {
      const progress = parseInt(Math.round((progressEvent.loaded * 100) / progressEvent.total));
      // Update state here
      onUploadProgress(progress);
    },
  });
}

*/
