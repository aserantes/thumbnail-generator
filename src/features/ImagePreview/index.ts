export { ImagePreview } from "./ImagePreview";

export {
  initialState,
  imagePreviewReducer,
  changeTest,
} from "./ImagePreviewSlice";
