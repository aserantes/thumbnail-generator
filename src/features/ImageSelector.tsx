import React from "react";
import styled from "@emotion/styled";

const Container = styled.div`
  background-color: orange;
  padding: 20px;
`;

export const ImageSelector = () => {
  return (
    <Container data-testid="ImageSelector">
      <div>ImageSelector Component</div>
    </Container>
  );
};
