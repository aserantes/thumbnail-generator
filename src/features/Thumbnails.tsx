import React from "react";
import styled from "@emotion/styled";

const Container = styled.div`
  background-color: yellow;
  padding: 20px;
`;

export const Thumbnails = () => {
  return (
    <Container data-testid="Thumbnails">
      <div>Thumbnails Component</div>
    </Container>
  );
};
