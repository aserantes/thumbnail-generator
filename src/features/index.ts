export { ImagePreview } from "./ImagePreview";

export { ImageSelector } from "./ImageSelector";

export { Thumbnails } from "./Thumbnails";
