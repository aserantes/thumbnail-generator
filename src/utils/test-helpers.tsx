import { render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
// TODO import the initial state of all slices, not just this example:
import { initialState as imagePreviewInitialState } from "store/ImagePreviewSlice";
import { RootState } from "store";

export const rootInitialState = {
  imagePreview: imagePreviewInitialState,
};

const mockStore = configureStore([thunk]);

export const renderWithRedux = (
  ui: JSX.Element,
  initialState: RootState = rootInitialState
) => {
  const store = mockStore(initialState);
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    mockStore: store,
  };
};

// https://png-pixel.com/
export const greenPixelBase64 =
  "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M/wHwAEBgIApD5fRAAAAABJRU5ErkJggg==";

export const redPixelBase64 =
  "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z8DwHwAFBQIAX8jx0gAAAABJRU5ErkJggg==";

/* export default function({ name = 'file.txt', size = 1024, type = 'plain/txt', lastModified = new Date() }) {
    const blob = new Blob(['a'.repeat(size)], { type });

    blob.lastModifiedDate = lastModified;

    return new File([blob], name);
}

const testFile = MockFile({
     type: 'image/png',
     size: 50000,
}); */

export * from "@testing-library/react";
