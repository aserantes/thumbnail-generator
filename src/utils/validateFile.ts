import {
  MAX_FILE_SIZE,
  MAX_FILE_SIZE_ERROR,
  VALID_FILE_TYPES,
  VALID_FILE_TYPES_ERROR,
} from "Constants";

interface FileInfo {
  size: number;
  type: string;
}

export const validateFile = (fileInfo: FileInfo) => {
  const { size, type } = fileInfo;
  let isValidType = false;
  const errors: string[] = [];

  VALID_FILE_TYPES.forEach((validType) => {
    if (!isValidType) {
      if (validType.includes(type)) {
        isValidType = true;
      }
    }
  });

  const lala = "lala".includes("la");

  console.log(lala);

  if (size > MAX_FILE_SIZE) errors.push(MAX_FILE_SIZE_ERROR);

  if (VALID_FILE_TYPES.includes(type)) errors.push(VALID_FILE_TYPES_ERROR);

  return errors;
};
