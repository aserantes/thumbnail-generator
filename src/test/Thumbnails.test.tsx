import React from "react";
import { render, screen } from "utils/test-helpers";
import { Thumbnails } from "features/Thumbnails";

describe("<Thumbnails>", () => {
  it("should render", () => {
    render(<Thumbnails />);
    const component = screen.getByTestId("Thumbnails");

    expect(component).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    expect(render(<Thumbnails />).asFragment()).toMatchSnapshot();
  });
});
