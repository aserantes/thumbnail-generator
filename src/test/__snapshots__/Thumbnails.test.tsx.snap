// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thumbnails> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-119czso"
    data-testid="Thumbnails"
  >
    <div>
      Thumbnails Component
    </div>
  </div>
</DocumentFragment>
`;
